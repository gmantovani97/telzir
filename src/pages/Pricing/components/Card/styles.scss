@import '../../../../styles/mixins.scss';
@import '../../../../styles/variables.scss';

@keyframes setShadow {
  from {
    box-shadow: 0 0 transparent;
  } to {
    box-shadow: 0 0 0.625rem $grey;
  }
}

.card {
  padding: 1.25rem;
  margin: 0 2.5rem;

  &--best {
    padding: 1.25rem;
    border: 0.125rem solid $primary-dark;
  }

  &:hover {
    animation: setShadow 500ms forwards;
  }

}

.flag {
  align-self: center;
  padding: 0.3125rem;
  height: 1.625rem;
  border-radius: 0.8125rem;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: $primary-color;

  &__icon {
    height: 1.25rem;
    fill: $white;
  }

  &__text {
    font-weight: 600;
    color: $white;
  }
}


.image {
  height: 4rem;
}

.title {
  align-self: center;
  margin: 1.25rem 0;
  letter-spacing: 0.0625rem;
  font-weight: 500;
  color: $secondary-dark;
}

.section {
  margin: 1.25rem 0;
  flex: 1;
  justify-content: center;

  &__item {
    flex-direction: row;
    margin: 0.3125rem 0;
  }

  &__icon {
    width: 1.25rem;
    margin-right: 0.3125rem;
    &--exclude {
      fill: #A00;;
    }
    &--include {
      fill: #0A0;
    }
  }

  &__text {
    font-size: 0.875rem;
  }

}

.price {
  justify-content: center;
  align-items: center;

  &__section {
    flex-direction: row;
    justify-content: center;
    align-items: center;
  }

  &__number {
    font-size: 1.25rem;
    margin-right: 0.125rem;
  }


  &__text {
    font-size: 0.875rem;
  }

}

@keyframes setBestButtonColor {
  from {
    background-color: $secondary-color;
    border: 0.125rem solid $secondary-color;
  } to {
    background-color: $secondary-dark;
    border: 0.125rem solid $secondary-dark;
  }
}

@keyframes setButtonColor {
  from {
    background-color: $white;
    border: 0.125rem solid $secondary-dark;
    p {
      color: $secondary-dark;
    }
  } to {
    background-color: $secondary-dark;
    border: 0.125rem solid $secondary-dark;
    p {
      color: $white;
    }
  }
}

@keyframes setButtonTextColor {
  from {
    color: $secondary-dark;
  } to {
    color: $white;
  }
}

@keyframes setBestButtonTextColor {
  to {
    color: $white;
  }
}



.button {
  background-color: $white;
  height: 3rem;
  justify-content: center;
  align-items: center;
  border-radius: 1.5rem;
  margin-top: 1.25rem;
  cursor: pointer;
  border: 0.125rem solid $secondary-dark;

  &:hover {
    animation: setButtonColor 500ms forwards;

    p {
      color: $white;
    }
  }

  &--best {
    background-color: $secondary-color;
    border: 0.125rem solid $secondary-color;

    &:hover {
      animation: setBestButtonColor 500ms forwards;
    }
  
  }

  &__text {
    color: $secondary-dark;
    font-weight: 800;
    
    &--best {
      color: $white;
    }
  }
}
