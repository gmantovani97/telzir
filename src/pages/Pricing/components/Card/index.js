import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import Icon from '@mdi/react';
import { mdiCheckBold, mdiClose, mdiSeal } from '@mdi/js';

import './styles.scss';

export default function Card({ best, data, choices }) {
  console.log(choices);
  console.log(data);
  const { maximumTime, tax, fee } = data;
  const { originValue, destinationValue, minutes } = choices;
  const exceededMinutes = maximumTime - minutes;
  const [price, setPrice] = useState({
    priceWithDiscount:
      exceededMinutes < 0 ? Math.abs(exceededMinutes) * (tax * fee) : 0,
    priceWithoutDiscount: maximumTime * tax,
  });

  useEffect(() => {
    setPrice({
      priceWithDiscount:
        exceededMinutes < 0 ? Math.abs(exceededMinutes) * (tax * fee) : 0,
      priceWithoutDiscount: mdiAccountMultiplePlusOutline * tax,
    });
    console.log(price);
  }, []);

  return (
    <div className={`card ${best && 'card--best'}`}>
      <img src={data.image} alt="Phone" className="image" />
      <h3 className="title">{data.name}</h3>
      {best && (
        <div className="flag">
          <Icon className="flag__icon" path={mdiSeal} />
          <p className="flag__text">Melhor plano</p>
        </div>
      )}
      <div className="section">
        <div className="section__item">
          <Icon
            className="section__icon section__icon--include"
            path={mdiCheckBold}
          />
          <p className="section__text">Ligações locais gratuitas</p>
        </div>
        <div className="section__item">
          <Icon
            className="section__icon section__icon--include"
            path={mdiCheckBold}
          />
          <p className="section__text">
            Ligações para mesma operadora sem custo
          </p>
        </div>
        <div className="section__item">
          <Icon
            className="section__icon section__icon--exclude"
            path={mdiClose}
          />
          <p className="section__text">
            Ligações entre 00:00 e 06:00 com 50% de desconto
          </p>
        </div>
        <div className="section__item">
          <Icon
            className="section__icon section__icon--exclude"
            path={mdiClose}
          />
          <p className="section__text">
            Ligações em feriado com 20% de desconto
          </p>
        </div>
      </div>
      <div className="price">
        <div className="price__section">
          <h3 className="price__number">82,50</h3>
          <h4 className="price__text">/mês</h4>
        </div>
      </div>
      <div className={`button ${best && 'button--best'}`}>
        <div className={`button__text ${best && 'button__text--best'}`}>
          Contratar
        </div>
      </div>
    </div>
  );
}

Card.propTypes = {
  best: PropTypes.bool,
  image: PropTypes.string.isRequired,
  data: PropTypes.shape.isRequired,
  choices: PropTypes.shape.isRequired,
};

Card.defaultProps = {
  best: false,
};
