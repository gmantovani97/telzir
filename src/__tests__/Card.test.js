import React from 'react';
import { render } from '@testing-library/react';

import Card from '@/pages/Pricing/components/Card';
import image from '@/assets/images/prime.svg';

describe('Card component', () => {
  it('matches snapshot', () => {
    const data = {
      id: 1,
      priceWithDiscount: '167.20',
      best: false,
      image,
      name: 'Falemais 120',
    };
    const { asFragment } = render(<Card data={data} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
