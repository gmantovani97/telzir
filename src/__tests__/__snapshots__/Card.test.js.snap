// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card false"
  >
    <div
      class="card__header"
    >
      <img
        alt="Phone"
        class="image"
        src="prime.svg"
      />
      <h3
        class="title"
      >
        Falemais 120
      </h3>
    </div>
    <div
      class="section"
    >
      <div
        class="section__item"
      >
        <svg
          class="section__icon section__icon--include"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
          />
        </svg>
        <p
          class="section__text"
        >
          Ligações locais gratuitas
        </p>
      </div>
      <div
        class="section__item"
      >
        <svg
          class="section__icon section__icon--include"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M9,20.42L2.79,14.21L5.62,11.38L9,14.77L18.88,4.88L21.71,7.71L9,20.42Z"
          />
        </svg>
        <p
          class="section__text"
        >
          Ligações para mesma operadora sem custo
        </p>
      </div>
      <div
        class="section__item"
      >
        <svg
          class="section__icon section__icon--exclude"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
          />
        </svg>
        <p
          class="section__text"
        >
          Ligações entre 00:00 e 06:00 com 50% de desconto
        </p>
      </div>
      <div
        class="section__item"
      >
        <svg
          class="section__icon section__icon--exclude"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
          />
        </svg>
        <p
          class="section__text"
        >
          Ligações em feriado com 20% de desconto
        </p>
      </div>
    </div>
    <div
      class="price"
    >
      <div
        class="price__section"
        data-testid="pricewithdiscount-1"
      >
        <h3
          class="price__number"
        >
          167,20
        </h3>
        <h4
          class="price__text"
        >
          /mês
        </h4>
      </div>
    </div>
    <div
      class="button false"
    >
      <p
        class="button__text false"
      >
        Contratar
      </p>
    </div>
  </div>
</DocumentFragment>
`;
